### PR TITLE
Splink4 ComparisonCreator ft. ExactMatch and LevenshteinAtThresholds

### PR DESCRIPTION
**Pull request by ADBond**
This is applying the same sort of idea as #1689 to fully-fledged `Comparison`s by introducing an abstract `ComparisonCreator` class which all of our `comparison_library` and `comparison_template_library` objects can inherit from. This is a stripped-back form missing quite a few features of the current classes, but enough to show the general shape of things.

Included:
* `ComparisonCreator` base class
* `comparison_library.ExactMatch`
* `comparison_library.LevenshteinAtThresholds`
* addition to the new temporary test module to cover using these in `settings` + running predict

Not included:
* any of the bells and whistles currently available for these library comparisons
* any mechanism analogous to `ComparisonLevelCreator.configure()`, as think it will be worth having a think about:
  * exactly what features we want to expose access to at this level rather than just building a custom comparison (individual `m_probability` values per level? `u_probability`?  term frequency stuff??)
  * what kind of sensible API we could have for providing access for that (using perhaps something more like `.configure()`), rather than having loads of arguments to individual comparisons in an ad-hoc manner as we currently have

Aside from the issue of dealing with multiple `col_name` here and in `ComparisonLevelCreator` that I mentioned elsewhere I _think_ this should be a suitable base to implement further `comparison_library` and `comparison_template_library` functions from.

### CODE REVIEW

**Review by RobinL — APPROVED** [review]
This looks great to me.  

There's a nice consistency with the comparison_level_creator in the names you've used, and overall the implementation is very straightforward - which I think is an indication we're on the right track with all our ideas so far.

I think you or someone else mentioned elsewhere, we probably want to think about what methods in comparison_level_creator and comaprison_creator are underscored (private).  I'll make a note on my todo list for us to do this in a separate PR 

Yeah - we now need to think about how to deal with multi-column comparison levels but good to get this merged as a starting point

File: splink/linker.py [hunk 1]
```diff
@@ -523,13 +526,16 @@ def _instantiate_comparison_levels(self, settings_dict):
...
         """
         dialect = self._sql_dialect
-        # TODO: handle case where dict is in fact a Comparison object
-        for comparison_dict in settings_dict["comparisons"]:
-            comparison_levels = comparison_dict["comparison_levels"]
-            for idx, level in enumerate(comparison_levels):
-                # if we have a ComparisonLevelCreator
-                if not isinstance(level, dict):
-                    comparison_levels[idx] = level.get_comparison_level(dialect)
+        comparisons = settings_dict["comparisons"]
```
Comment (RobinL) [inline splink/linker.py:529]: This key is not guaranteed to exist
```suggestion
        if "comparisons" not in settings_dict:
            return
        comparisons = settings_dict["comparisons"]
```

**Comment by RobinL** [comment]
Super-simple testing script:

<details>

```python
from splink.comparison_level_library import NullLevel
from splink.comparison_library import ExactMatch
from splink.datasets import splink_datasets
from splink.duckdb.linker import DuckDBLinker

NullLevel("surname").get_comparison_level("duckdb").as_dict()

df = splink_datasets.fake_1000
settings = {
    "probability_two_random_records_match": 0.01,
    "link_type": "dedupe_only",
    "blocking_rules_to_generate_predictions": [],
    "comparisons": [
        ExactMatch("surname"),
    ],
}


linker = DuckDBLinker(df, settings)
linker.predict().as_pandas_dataframe()

settings = {
    "probability_two_random_records_match": 0.01,
    "link_type": "dedupe_only",
    "blocking_rules_to_generate_predictions": [],

}


linker = DuckDBLinker(df, settings)

```

</details>

File: splink/linker.py (hunk 1 above)
Comment (ADBond) [reply splink/linker.py:529]: Ah yep, good shout. Yeah this bit of the code will definitely need revising / some more careful thought - just had something simple that mostly works, as we may end up changing how some of this linker code works more generally

**Pull request merged**